specification/08-attestation.adoc

```diff
@@ -68,6 +75,74 @@ It always collects the up-to-date endorsement and RIM from the device relying pa
 | Verifier Location               | Inside of TVM         | Inside of TVM or Remove Service
 |===
 
+==== Device Runtime Update and Reattestation
+
+A TEE-IO device may support runtime update. Currently, there are two policy
+settings controlling device runtime update.
+
+1. SPDM Session `TerminationPolicy`
+
+It controls if runtime update will cause SPDM session termination or device may
+keep the session. NOTE: The session termination means the IDE stream will
+change to `insecure` state and TDI in `ERROR` state.
+
+The SPDM `TerminationPolicy` is set in SPDM `KEY_EXCHANGE` message by
+TSM. The TSM should return such policy information to the TVM via
+`sbi_covg_get_device_spdm_attrs()`.
+
+2. TDISP `NO_FW_UPDATE`
+
+It controls if runtime update is allowed after TDI in `CONFIG_LOCKED` or `RUN`
+state. NOTE: Runtime update is always allowed if TDI is in `CONFIG_UNLOCKED`
+or `ERROR`.
+
+The TDISP `NO_FW_UPDATE` is set in TDISP `LOCK_INTERFACE_REPORT` message by TSM,
+and is returned in BIT0 of `INTERFACE_INFO` in `TDI Report Structure` by
+the device. The TVM can get the full `TDI Report Structure` via
+`sbi_covg_get_interface_report()`.
+
+The TVM should consult both setting to decide if it can accept the device
+runtime update.
+
+If the TVM allows the device runtimea after lock (`NO_FW_UPDATE`=0) and update
+without session termination (`TerminationPolicy`=1), then the TVM may want to
+do attestation again to verify the device. There are 2 possible models.
+
+1. Post-update reattestation
+
+After the device performs the update, TVM will collect certificate and
+measurement again and do verification again to determine if the TVM still wants
+to accept the device. If the TVM decides to not accept the device any more, the
+TVM will communicate with VMM and ask for `sbi_covh_unbind_interface()`, then
+verify with TSM if the TDI is really unbound by the VMM.
+
+How the device performs the update is out of scope of this specification.
+It could be initiated by current TVM, other TVMs, other legacy VMs, VMM, or even
+out of band (OOB) mechanism, such as Baseboard Management Controller (BMC).
+
+When the TVM performs the reattesttion is also out of scope of this
+specification. It could be a pull mode with policy inside of TVM, such as
+reattestation every week. Or it could be an interrupt mode such as request
+triggered by the orchestrator.
+
+2. Pre-update attestation
+
+With post-update reattestation, there is a gap between update time and
+attestation time. Since PCIe TDISP requires the secure device firmware update,
```

Comment:
```suggestion
With post-update reattestation, there is a gap between update time and
attestation time. Confidential data access from the TDI may happen before the updated device is re-attested and re-accepted by the TVM.

Since PCIe TDISP requires the secure device firmware update,
```

Reply:
done